src/PluginManager.cpp

```diff
@@ -18,1064 +18,23 @@ effects, generators, analysis-effects, commands.  It also has functions
-
-   auto msg = XO("Enabling effects or commands:\n\n%s").Format( last3 );
-
-   // Make sure the progress dialog is deleted before we call EndModal() or
-   // we will leave the project window in an unusable state on OSX.
-   // See bug #1192.
-   {
-      ProgressDialog progress{
-         Verbatim( GetTitle() ), msg, pdlgHideStopButton };
-      progress.CenterOnParent();
-
-      int i = 0;
-      for (ItemDataMap::iterator iter = mItems.begin(); iter != mItems.end(); ++iter)
-      {
-         ItemData & item = iter->second;
-         wxString path = item.path;
-
-         if (item.state == STATE_Enabled && item.plugs[0]->GetPluginType() == PluginTypeStub)
-         {
-            last3 = last3.AfterFirst(wxT('\n')) + item.path + wxT("\n");
-            auto status = progress.Update(++i, enableCount,
-               XO("Enabling effect or command:\n\n%s").Format( last3 ));
-            if (status == ProgressResult::Cancelled)
-            {
-               break;
-            }
-
-            TranslatableString errMsgs;
-
-            // Try to register the plugin via each provider until one succeeds
-            for (size_t j = 0, cntj = item.plugs.size(); j < cntj; j++)
-            {
-               TranslatableString errMsg;
-               if (mm.RegisterEffectPlugin(item.plugs[j]->GetProviderID(), path,
-                                     errMsg))
-               {
-                  for (size_t k = 0, cntk = item.plugs.size(); k < cntk; k++)
-                  {
-                     pm.mPlugins.erase(item.plugs[k]->GetProviderID() + wxT("_") + path);
-                  }
-                  // Bug 1893.  We've found a provider that works.
-                  // Error messages from any that failed are no longer useful.
-                  errMsgs = {};
-                  break;
-               }
-               else
-               {
-                  if (!errMsgs.empty())
-                     errMsgs.Join( errMsg, '\n' );
-                  else
-                     errMsgs = errMsg;
-               }
-            }
-            if (!errMsgs.empty())
-               AudacityMessageBox(
-                  XO("Effect or Command at %s failed to register:\n%s")
-                     .Format( path, errMsgs ) );
-         }
-         else if (item.state == STATE_New)
-         {
-            for (size_t j = 0, cnt = item.plugs.size(); j < cnt; j++)
-            {
-               item.plugs[j]->SetValid(false);
-            }
-         }
-         else if (item.state != STATE_New)
-         {
-            for (size_t j = 0, cnt = item.plugs.size(); j < cnt; j++)
-            {
-               item.plugs[j]->SetEnabled(item.state == STATE_Enabled);
-               item.plugs[j]->SetValid(item.valid);
-            }
-         }
-      }
-
-      pm.Save();
-   }
-
-   EndModal(wxID_OK);
-}
-
-void PluginRegistrationDialog::OnCancel(wxCommandEvent & WXUNUSED(evt))
-{
-   EndModal(wxID_CANCEL);
-}
-
-
```

Comment:
Just a lot of cut-and-paste in this commit.

But is there any tool that makes it easy "diff" the deleted blocks in this column, against the added blocks in the other, so you can check that I made no mistakes?  (Asking, I don't know)

Reply:
Agreed, that's it's not trivial to compare such blocks in that way, and if it's only a copy-paste (not copy-paste-modify) then, i believe, there are no mistakes. The only way i can come up with to compare such changes is to copy both of these blocks to the local files and compare them using a diff tool

Reply:
I think now that to prepare the branch in a way that makes verification easier, I could make one commit that duplicates the file into the new file path, but does not compile it.  The high similarity index of the file could be recognized by the diff viewer.

Then I make a second commit that compiles the new file, with only deletions of parts from the old and new files, and any other necessary changes of names and paths.